File: codingjs/scoresAverage.js
function scoresAverage(scores: number[]): number {
  if (scores.length === 2) return scores[0] > scores[1] ? scores[0] : scores[1]
  const halfIndex = scores.length / 2;
  const firstHalfAvg = getArrSum(scores.slice(0, halfIndex), 0, 0);
  const secondHalfAvg = getArrSum(scores.slice(halfIndex), 0, 0);
  return firstHalfAvg > secondHalfAvg ? firstHalfAvg : secondHalfAvg;
}
function getArrSum(halfScores: number[], index: number, sum: number): number {
  if (index >= halfScores.length) return sum / index
  return getArrSum(halfScores, ++index, sum + halfScores[index - 1]);
}

File: codingbat/java/scoresAverage.java
public int scoresAverage(int[] scores) {
  // if (scores.length < 2) return false;
  int centerPos = scores.length / 2;
  int firstHalfAvg = average(scores, 0, centerPos);
  int secondHalfAvg = average(scores, centerPos, scores.length);
  return firstHalfAvg > secondHalfAvg ? firstHalfAvg : secondHalfAvg;
}

public int average(int[] scores, int start, int end) {
  int sumNum = 0;
  for (int i = start; i < end; i++) {
    sumNum += scores[i];
  }
  return sumNum / (end - start);
}

//(end - start) in line 14 is tricky
